client/src/components/layouts/default.tsx

```diff
@@ -100,55 +101,51 @@ const getSystemTheme = () =>
       : 'light-palette'
   }`;
 
-class DefaultLayout extends Component<DefaultLayoutProps> {
-  static displayName = 'DefaultLayout';
-
-  componentDidMount() {
-    const { isSignedIn, fetchUser } = this.props;
+function DefaultLayout({
+  children,
+  hasMessage,
+  fetchState,
+  flashMessage,
+  isOnline,
+  isServerOnline,
+  isSignedIn,
+  removeFlashMessage,
+  showFooter = true,
+  isChallenge = false,
```

Comment:
if we to decide values for these props, maybe it's better to have them strict and not `undefined`, so it doesn't become weird trying to figure out where is the value assigned.

But this outside the scope of this PR.